test/e2e/alertmanager_e2e_test.go

```diff
@@ -79,3 +83,111 @@ func TestAlertmanagerVersionMigration(t *testing.T) {
+		t.Fatal(err)
+	}
+
+	resp, err := http.Get(fmt.Sprintf("http://%s:30903/", framework.ClusterIP))
+	if err != nil {
+		t.Fatal("Retrieving alertmanager landing page failed with error: ", err)
+	} else if resp.StatusCode != 200 {
+		t.Fatal("Retrieving alertmanager landing page failed with http status code: ", resp.StatusCode)
+	}
+}
+
+func TestExposingAlertmanagerWithKubernetesAPI(t *testing.T) {
+	alertmanager := framework.MakeBasicAlertmanager("test-alertmanager", 1)
+	alertmanagerService := framework.MakeAlertmanagerService(alertmanager.Name, "alertmanager-service", v1.ServiceTypeClusterIP)
+
+	defer func() {
+		if err := framework.DeleteAlertmanagerAndWaitUntilGone(alertmanager.Name); err != nil {
+			t.Fatal(err)
+		}
+		if err := framework.KubeClient.CoreV1().Services(framework.Namespace.Name).Delete(alertmanagerService.Name, nil); err != nil {
+			t.Fatal(err)
+		}
+	}()
+
+	if err := framework.CreateAlertmanagerAndWaitUntilReady(alertmanager); err != nil {
+		t.Fatal(err)
+	}
+
+	if _, err := framework.KubeClient.CoreV1().Services(framework.Namespace.Name).Create(alertmanagerService); err != nil {
+		t.Fatal(err)
+	}
+
+	if err := framework.WaitForServiceReady(time.Minute, alertmanagerService.Name); err != nil {
+		t.Fatal(err)
+	}
+
+	ProxyGet := framework.KubeClient.CoreV1().Services(framework.Namespace.Name).ProxyGet
+	request := ProxyGet("", alertmanagerService.Name, "web", "/", make(map[string]string))
+	_, err := request.DoRaw()
+	if err != nil {
+		t.Fatal(err)
+	}
+}
+
+func TestExposingAlertmanagerWithIngress(t *testing.T) {
+	alertmanager := framework.MakeBasicAlertmanager("main", 1)
+	alertmanagerService := framework.MakeAlertmanagerService(alertmanager.Name, "test-group", v1.ServiceTypeClusterIP)
+	ingress := framework.MakeBasicIngress(alertmanagerService.Name, 9093)
+
+	defer func() {
+		if err := framework.DeleteAlertmanagerAndWaitUntilGone(alertmanager.Name); err != nil {
+			t.Fatal(err)
+		}
+		if err := framework.KubeClient.CoreV1().Services(framework.Namespace.Name).Delete(alertmanagerService.Name, nil); err != nil {
+			t.Fatal(err)
+		}
+		if err := framework.KubeClient.Extensions().Ingresses(framework.Namespace.Name).Delete(ingress.Name, nil); err != nil {
+			t.Fatal(err)
+		}
+		if err := framework.DeleteNginxIngressControllerIncDefaultBackend(); err != nil {
+			t.Fatal(err)
+		}
+	}()
+
+	if err := framework.SetupNginxIngressControllerIncDefaultBackend(); err != nil {
+		t.Fatal(err)
+	}
+
+	if err := framework.CreateAlertmanagerAndWaitUntilReady(alertmanager); err != nil {
+		t.Fatal(err)
+	}
+
+	if _, err := framework.KubeClient.CoreV1().Services(framework.Namespace.Name).Create(alertmanagerService); err != nil {
+		t.Fatal(err)
+	}
+
+	if err := framework.CreateIngress(ingress); err != nil {
+		t.Fatal(err)
+	}
+
+	err := framework.WaitForHTTPSuccessStatusCode(time.Second*30, fmt.Sprintf("http://%s:80/metrics", framework.ClusterIP))
```

Comment:
Why specify port 80 explicitly here? If it's not required I would skip it as it makes it seem like it is.